pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ExpressionFunctions.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function.scalar;
+
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+public class ExpressionFunctions {
+  private ExpressionFunctions() {
+  }
+
+  @ScalarFunction(vargsFunction = true)
+  public static boolean and(Boolean... args) {
+    for (Boolean arg : args) {
+      if (!arg) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
This function is to support logical conjugation of predicates, but is it the right approach? This requires all expressions to be evaluated before they are combined, and prevents early termination. Early termination is not just a performance concern, but one of flexibility and correctness - it should be possible to express predicates which aren't safe to evaluate if another predicate acts as a guard to prevent its evaluation.

Reply:
+1 on this. Will rework this to allow early termination.

Reply:
Have tried to model these functions differently (to be able to support early termination) here https://github.com/apache/pinot/pull/8582/files